plugin/trino-mariadb/src/main/java/io/trino/plugin/mariadb/MariaDbJdbcConfig.java

```diff
@@ -29,19 +26,4 @@ public boolean isUrlValid()
         Driver driver = new Driver();
         return driver.acceptsURL(getConnectionUrl());
     }
-
-    @AssertTrue(message = "Database (catalog) must not be specified in JDBC URL for MariaDB connector")
```

Comment:
This check exists for #12712

Reply:
It sounds like the only way to fix properly this is to implement your idea on #12712:

> we can avoid the above failure by adjusting the logic around quoting the identifier

What code do you think would need to be fixed to address this? I can take a look.

Since #12712 is only a problem in certain situations (such as creating a table), the other option would be to add an additional configuration parameter to the connector, such as `enable-proxysql-workaround=true`, which could bypass this check.

Reply:
> What code do you think would need to be fixed to address this?

I'm not sure. I would recommend testing all statements.